### DISCUSSION TITLE
Only include on the bundle the fonts that are actually in use

### DISCUSSION THREAD

**Discussion by tecsobs** [vercel/next.js#44808] [Ideas]
### Describe the feature you'd like to request

With react + tailwindcss we can create multiple @font-face with different weight for a font family and only the fonts that have been used are included in the app. If in the app we only use font-light and font-medium only those files are requested.

With tailwindcss will be something like this

@font-face {
  font-family: Poppins;
  font-weight: 300;
  src: url("../fonts/Poppins/Poppins-Light.ttf");
}
@font-face {
  font-family: Poppins;
  font-weight: 400;
  src: url("../fonts/Poppins/Poppins-Regular.ttf");
}
@font-face {
  font-family: Poppins;
  font-weight: 500;
  src: url("../fonts/Poppins/Poppins-Medium.ttf");
}
@font-face {
  font-family: Poppins;
  font-weight: 600;
  src: url("../fonts/Poppins/Poppins-SemiBold.ttf");
}
@font-face {
  font-family: Poppins;
  font-weight: 700;
  src: url("../fonts/Poppins/Poppins-Bold.ttf");
}

But in the case below all font files are requested in the app even when not all are used

import localFont from '@next/font/local'

const poppins = localFont({
    src: [
        {
            path: './Poppins/Poppins-Light.ttf',
            weight: '300',
        },
        {
            path: './Poppins/Poppins-Regular.ttf',
            weight: '400',
        },
        {
            path: './Poppins/Poppins-Medium.ttf',
            weight: '500',
        },
        {
            path: './Poppins/Poppins-SemiBold.ttf',
            weight: '600',
        },
        {
            path: './Poppins/Poppins-Bold.ttf',
            weight: '700',
        },
    ],
    variable: '--font-display',
})

### Describe the solution you'd like

@next/font should only send to the client the fonts being used like with tailwindcss and not all of them

### Describe alternatives you've considered

@next/font should only send to the client the fonts being used like with tailwindcss and not all of them

**Comment by hanneslund** [comment]
You could try disable [preloading](https://nextjs.org/docs/api-reference/next/font#preload) and setting [display](https://nextjs.org/docs/api-reference/next/font#display) to `swap`.
